apps/dashboard/src/contract-ui/tabs/nfts/components/claim-button.tsx

```diff
@@ -15,26 +32,133 @@ interface NFTClaimButtonProps {
-        onClose={onClose}
-        isOpen={isOpen}
-      >
-        <NFTClaimForm contract={contract} />
-      </Drawer>
-      <Button
-        colorScheme="primary"
-        leftIcon={<GemIcon className="size-4" />}
-        onClick={onOpen}
-      >
-        Claim
-      </Button>
-    </>
+    <Sheet open={open} onOpenChange={setOpen}>
+      <SheetTrigger asChild>
+        <Button colorScheme="primary" leftIcon={<GemIcon className="size-4" />}>
+          Claim
+        </Button>
+      </SheetTrigger>
+      <SheetContent className="z-[10000] overflow-y-auto sm:w-[540px] sm:max-w-[90%] lg:w-[700px]">
+        <SheetHeader>
+          <SheetTitle>Claim NFTs</SheetTitle>
+        </SheetHeader>
+        <form className="mt-8 flex w-full flex-col gap-3 md:flex-row">
+          <div className="flex w-full flex-col gap-6 md:flex-row">
+            <FormControl isRequired isInvalid={!!errors.to}>
+              <FormLabel>To Address</FormLabel>
+              <Input placeholder={ZERO_ADDRESS} {...register("to")} />
+              <FormHelperText>Enter the address to claim to.</FormHelperText>
+              <FormErrorMessage>{errors.to?.message}</FormErrorMessage>
+            </FormControl>
+            <FormControl isRequired isInvalid={!!errors.amount}>
+              <FormLabel>Amount</FormLabel>
+              <Input
+                type="text"
+                {...register("amount", {
+                  validate: (value) => {
+                    const valueNum = Number(value);
+                    if (!Number.isInteger(valueNum)) {
+                      return "Amount must be an integer";
+                    }
+                  },
+                })}
+              />
+              <FormHelperText>How many would you like to claim?</FormHelperText>
+              <FormErrorMessage>{errors.amount?.message}</FormErrorMessage>
+            </FormControl>
+          </div>
+        </form>
+        <div className="mt-4 flex justify-end">
+          <TransactionButton
+            txChainID={contract.chain.id}
+            transactionCount={1}
+            form={CLAIM_FORM_ID}
+            isLoading={formState.isSubmitting}
+            type="submit"
+            colorScheme="primary"
+            onClick={handleSubmit(async (d) => {
+              trackEvent({
+                category: "nft",
+                action: "claim",
+                label: "attempt",
+              });
+              if (!account) {
+                return toast.error("No account detected");
+              }
+              if (!d.to) {
+                return toast.error(
+                  "Please enter the address that will receive the NFT",
+                );
+              }
+
+              const transaction = claimTo({
+                contract,
+                to: d.to,
+                quantity: BigInt(d.amount),
+                from: account.address,
+              });
+
+              const approveTx = await getApprovalForTransaction({
```

Comment:
This can throw error - it should be wrapped in try-catch ( as it was earlier ) -wrap the entire submit handler with try catch